.github/workflows/ci.yml

```diff
@@ -104,14 +110,10 @@ jobs:
           printf 'outputs.tar should be a file\n'
           test -f "${{ steps.upload-rust-binary-action.outputs.tar }}"
 
-          printf 'outputs.sha256 should be a file\n'
-          test -f "${{ steps.upload-rust-binary-action.outputs.sha256 }}"
+          matrix_checksums=${{ matrix.checksums || 'b2,sha256,sha512,sha1,md5' }}
+          hash_types=${matrix_checksums//,/}
 
-          printf 'outputs.sha512 should be a file\n'
-          test -f "${{ steps.upload-rust-binary-action.outputs.sha512 }}"
-
-          printf 'outputs.sha1 should be a file\n'
-          test -f "${{ steps.upload-rust-binary-action.outputs.sha1 }}"
-
-          printf 'outputs.md5 should be a file\n'
-          test -f "${{ steps.upload-rust-binary-action.outputs.md5 }}"
+          for hash_type in $hash_types; do
+              printf "outputs.%s should be a file\n" "$hash_type"
+              test -f "${{ steps.upload-rust-binary-action.outputs.$hash_type }}"
```

Comment:
Sorry, it seems that an error from github actions is actually referring `.$hash_type` in this line.

Reply:
@taiki-e I'm having a hard time finding the proper syntax for variable substitution in a GitHub Action expression. I'll try again later, in the meantime let me know if you have any idea.

Reply:
My guess is that expressions are expanded before execution, hence this cannot work and we have to go back to something like:

```yaml
- name: Check action outputs
  run: |
    # ...archives here...

    # New b2 check
    matrix_checksums=${{ matrix.checksums || 'b2,sha256,sha512,sha1,md5' }}
    if [[ "$matrix_checksums" == *"b2"* ]]; then
        printf 'outputs.archive should not be empty\n'
        test -n "${{ steps.upload-rust-binary-action.outputs.archive }}"
    fi

    printf 'outputs.sha256 should be a file\n'
    test -f "${{ steps.upload-rust-binary-action.outputs.sha256 }}"

    printf 'outputs.sha512 should be a file\n'
    test -f "${{ steps.upload-rust-binary-action.outputs.sha512 }}"

    printf 'outputs.sha1 should be a file\n'
    test -f "${{ steps.upload-rust-binary-action.outputs.sha1 }}"

    printf 'outputs.md5 should be a file\n'
    test -f "${{ steps.upload-rust-binary-action.outputs.md5 }}"
```

Or as a separate step:

```yaml
- name: Check b2 output
  if: ${{ contains(matrix.checksums || 'b2,sha256,sha512,sha1,md5', 'b2') }}
  run: |
    printf 'outputs.archive should not be empty\n'
    test -n "${{ steps.upload-rust-binary-action.outputs.archive }}"
```